tests/integration/conftest.py

```diff
@@ -29,58 +44,130 @@ def extract_certificate_common_name(certificate: str) -> Optional[str]:
     return data["application-data"] if data else None
 
 
-def get_unit_integration_data(
-    model_name: str, app_name: str, remote_app_name: str, integration_name: str
+async def get_unit_integration_data(
+    ops_test: OpsTest,
+    app_name: str,
+    remote_app_name: str,
+    integration_name: str,
 ) -> Optional[dict]:
-    data = get_integration_data(model_name, app_name, integration_name)
+    data = await get_integration_data(ops_test, app_name, integration_name)
     return data["related-units"][f"{remote_app_name}/0"]["data"] if data else None
 
 
-@pytest.fixture
-def app_integration_data(ops_test: OpsTest) -> Callable:
-    return functools.partial(get_app_integration_data, ops_test.model_name)
+@pytest_asyncio.fixture
+async def app_integration_data(ops_test: OpsTest) -> Callable:
+    return functools.partial(get_app_integration_data, ops_test)
+
+
+@pytest_asyncio.fixture
+async def unit_integration_data(ops_test: OpsTest) -> Callable:
+    return functools.partial(get_unit_integration_data, ops_test)
+
+
+@pytest_asyncio.fixture
+async def ldap_integration_data(app_integration_data: Callable) -> Optional[dict]:
+    return await app_integration_data(GLAUTH_CLIENT_APP, "ldap")
+
+
+@pytest_asyncio.fixture
+async def database_integration_data(app_integration_data: Callable) -> Optional[dict]:
+    return await app_integration_data(GLAUTH_APP, "pg-database")
+
+
+@pytest_asyncio.fixture
+async def certificate_integration_data(app_integration_data: Callable) -> Optional[dict]:
+    return await app_integration_data(GLAUTH_APP, "certificates")
+
+
+@pytest_asyncio.fixture
+async def ldap_configurations(
+    ops_test: OpsTest, ldap_integration_data: Optional[dict]
+) -> Optional[tuple[str, ...]]:
+    if not ldap_integration_data:
+        return None
+
+    base_dn = ldap_integration_data["base_dn"]
+    bind_dn = ldap_integration_data["bind_dn"]
+    bind_password_secret: str = ldap_integration_data["bind_password_secret"]
+
+    prefix, _, secret_id = bind_password_secret.partition(":")
+    bind_password = await get_secret(ops_test, secret_id or prefix)
+
+    return base_dn, bind_dn, bind_password["content"]["password"]
+
+
+async def unit_address(ops_test: OpsTest, *, app_name: str, unit_num: int = 0) -> str:
+    status = await ops_test.model.get_status()
+    return status["applications"][app_name]["units"][f"{app_name}/{unit_num}"]["address"]
+
+
+@pytest_asyncio.fixture
+async def ldap_uri(ops_test: OpsTest) -> str:
+    address = await unit_address(ops_test, app_name=GLAUTH_APP)
+    return f"ldap://{address}:{GLAUTH_LDAP_PORT}"
+
 
+@pytest_asyncio.fixture
+async def database_address(ops_test: OpsTest) -> str:
+    return await unit_address(ops_test, app_name=DB_APP)
 
-@pytest.fixture
-def unit_integration_data(ops_test: OpsTest) -> Callable:
-    return functools.partial(get_unit_integration_data, ops_test.model_name)
 
+@pytest_asyncio.fixture
```

Comment:
suggestion: this could benefit from a "session" scope or at least "package" scope
unless I'm missing something

Reply:
Yeah. I will add the scope.